### ISSUE TITLE
running two schema changes at the same time?

### ISSUE THREAD

**Issue by dianasaur323** [cockroachdb/cockroach#17874]
**QUESTION**

My understanding was that we only allowed one schema change to run at a time.

I am on the PM cluster. I imported TPC-H scalefactor 1 using restore. I ran ```ALTER TABLE customer ADD COLUMN names STRING;```, then subsequently ran ```ALTER TABLE customer ADD COLUMN names2 STRING;```

I would expect to not be able to run both at the same time, but I'm seeing both of them executing when I run ```SHOW QUERIES```

This is what ```SHOW QUERIES``` looks like: 
![image](https://user-images.githubusercontent.com/5768024/29639264-29dc7bf8-8828-11e7-807c-5a097624e1ab.png)

After waiting for a couple minutes, the jobs are now showing up, but I don't believe they are running. 

![image](https://user-images.githubusercontent.com/5768024/29639414-b3499eac-8828-11e7-87cf-fcae0a8db2a0.png)

They do still appear to be executing in ```SHOW QUERIES```

After waiting for a while, it appears that there is some significant lag between when a query is executed and when it shows up in the jobs table. The schema changes correctly only ran one at a time, but the issue remains somehow what shows up in SHOW JOBS doesn't reflect what shows up in SHOW QUERIES.

cc @jordanlewis and @benesch

**Comment by jordanlewis** [comment]
Only one of the two will actually get executed at once, but they will both show up up in `SHOW JOBS`. AFAIK neither should show up in `SHOW QUERIES`, and I believe @benesch has already fixed that.

**Comment by dianasaur323** [comment]
I'm not seeing it in ```SHOW JOBS``` This should be from the most recent master branch. Does this mean perhaps that @benesch's changes haven't made it in yet?

**Comment by bdarnell** [comment]
Adding a column (with no default) is fast; it doesn't need to create a backfill job. Maybe something is inserting it into the jobs table but because there's no real job it's not cleaning up?

**Comment by benesch** [comment]
I think @itsbilal was working on hiding job-creating queries from `SHOW QUERIES`.

**Comment by dianasaur323** [comment]
Hmm.. that's what people have been saying, but looking at the time it took to complete adding a column with no default, it seems like I probably should have observed a progress bar at some point.

![image](https://user-images.githubusercontent.com/5768024/29640149-84bd46d0-882b-11e7-9e54-35803732d18b.png)

However, I only saw 0% and then Succeeded. Perhaps I wasn't staring at it closely enough.  

![image](https://user-images.githubusercontent.com/5768024/29640130-72f8afa2-882b-11e7-9dea-163a2ecae5c1.png)

I just tested out another command: ALTER TABLE customer DROP COLUMN names2, and that one successfully showed up in the Jobs Table pretty quickly, and is now showing the progress bar. I'm starting to wonder if that was a fluke, and or if somehow adding null columns are treated differently. I could try to repro it tomorrow and see if the problems till occurs.
